osf/models/preprint.py

```diff
@@ -315,18 +320,32 @@ def create_version(cls, dupliate_from_guid):
         data_for_update['custom_publication_citation'] = source_preprint.custom_publication_citation
         data_for_update['article_doi'] = source_preprint.article_doi
 
+        data_for_update['has_coi'] = source_preprint.has_coi
+        data_for_update['conflict_of_interest_statement'] = source_preprint.conflict_of_interest_statement
         data_for_update['has_data_links'] = source_preprint.has_data_links
+        data_for_update['why_no_data'] = source_preprint.why_no_data
         data_for_update['data_links'] = source_preprint.data_links
         data_for_update['has_prereg_links'] = source_preprint.has_prereg_links
+        data_for_update['why_no_prereg'] = source_preprint.why_no_prereg
         data_for_update['prereg_links'] = source_preprint.prereg_links
 
+        if source_preprint.node:
+            data_for_update['node'] = source_preprint.node
+
         preprint = cls(
             provider=source_preprint.provider,
             title=source_preprint.title,
             creator=source_preprint.creator,
             description=source_preprint.description,
         )
         preprint.save()
+        # add contributors
+        for contributor_info in source_preprint.contributor_set.exclude(user=source_preprint.creator).values('visible', 'user_id', '_order'):
+            preprint.contributor_set.create(**{**contributor_info, 'preprint_id': preprint.id})
+
+        # add affiliated institutions
+        for institution in source_preprint.affiliated_institutions.all():
+            preprint.add_affiliated_institution(institution, auth.user)
```

Comment:
Preprint affiliation has a trick, we can take a look together on Monday. Basically, user can only add it's own affiliate institution instead. Thus, this will fail if one of the affiliation is added by another contributor and is not affiliate with the version creator. I think we need a helper method to add affiliation bypass user affiliation check for this.